### PR TITLE
[improvement](routine-load) Make routine load work with old kafka version

### LINKED ISSUE #7629: [Enhancement] Make routine load works with old kafka version
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

We create routine job and then failed, the fatal error log is like following:
I0105 10:53:46.847640 23683 data_consumer.h:86] kafka error: Local: Broker transport failure, event: nameserver.jmq.jd.local:80/bootstrap: ApiVersionRequest failed: Local: Bad message format: probably due to broker version < 0.10 (see api.version.request configuration) (after 0ms in state APIVERSION_QUERY, 1 identical error(s) suppressed)

this is because we use kafka with old version which not be compatible with librdkafka we used now.

### Solution

add config to make routine load works with old kafka version

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by caiconghui**
## Proposed changes

Close related #7629  (replace it with issue number if it exists).

Describe the overview of changes, and introduce why we need it.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [x] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #7629) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 1]
```diff
@@ -75,8 +75,9 @@ Status KafkaDataConsumer::init(StreamLoadContext* ctx) {
     // TODO: set it larger than 0 after we set rd_kafka_conf_set_stats_cb()
     RETURN_IF_ERROR(set_conf("statistics.interval.ms", "0"));
     RETURN_IF_ERROR(set_conf("auto.offset.reset", "error"));
-    RETURN_IF_ERROR(set_conf("api.version.request", "true"));
+    RETURN_IF_ERROR(set_conf("api.version.request", config::kafka_api_version_request));
```
Comment (EmmyMiao87) [inline be/src/runtime/routine_load/data_consumer.cpp:78]: Is It correct when api.version.request=true and broker.version.fallback=0.9 for new version of libdfkafka

Comment (caiconghui) [reply be/src/runtime/routine_load/data_consumer.cpp:78]: image
I will set broker.version.fallback=0.10.0 for default

Comment (EmmyMiao87) [reply be/src/runtime/routine_load/data_consumer.cpp:78]: Great ~

File: docs/en/administrator-guide/config/be_config.md [hunk 2]
```diff
@@ -678,6 +678,18 @@ Default：10737418240
 
 BloomFilter/Min/Max and other statistical information cache capacity
 
+### `kafka_api_version_request`
```
Comment (EmmyMiao87) [inline docs/en/administrator-guide/config/be_config.md:681]: Could you please update Routine Load doc ? https://doris.apache.org/master/zh-CN/administrator-guide/load-data/routine-load-manual.html#%E5%8E%9F%E7%90%86

Comment (caiconghui) [reply docs/en/administrator-guide/config/be_config.md:681]: done

File: docs/zh-CN/administrator-guide/load-data/routine-load-manual.md [hunk 3]
```diff
@@ -85,7 +85,7 @@ FE 中的 JobScheduler 根据汇报结果，继续生成后续新的 Task，或
 
 1. 支持无认证的 Kafka 访问，以及通过 SSL 方式认证的 Kafka 集群。
 2. 支持的消息格式为 csv, json 文本格式。csv 每一个 message 为一行，且行尾**不包含**换行符。
-3. 仅支持 Kafka 0.10.0.0(含) 以上版本。
+3. 仅支持 Kafka 0.10.0.0(含) 以上版本。如果要使用 Kafka 0.10.0.0 以下版本 (0.9.0, 0.8.2, 0.8.1, 0.8.0)，需要修改 be 的配置，将 kafka_api_version_request 的值设置为false，将 kafka_broker_version_fallback 的值设置为要兼容的旧版本，使用旧版本的代价是routine load 的部分新特性无法使用，如根据时间设置 kafka 分区的 offset。
```
Comment (EmmyMiao87) [inline docs/zh-CN/administrator-guide/load-data/routine-load-manual.md:88]:
```suggestion
3. 默认支持 Kafka 0.10.0.0(含) 以上版本。如果要使用 Kafka 0.10.0.0 以下版本 (0.9.0, 0.8.2, 0.8.1, 0.8.0)，需要修改 be 的配置，将 kafka_api_version_request 的值设置为false，将 kafka_broker_version_fallback 的值设置为要兼容的旧版本，使用旧版本的代价是routine load 的部分新特性无法使用，如根据时间设置 kafka 分区的 offset。
```

Comment (caiconghui) [reply docs/zh-CN/administrator-guide/load-data/routine-load-manual.md]: done

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**